conans/test/integration/package_info_test.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+
+from conans.paths import CONANFILE, CONANFILE_TXT
+from conans.test.utils.tools import TestClient
+
+
+class TestPackageInfo(unittest.TestCase):
+
+    def package_info_called_in_local_cache_test(self):
+        client = TestClient()
+        conanfile_tmp = '''
+from conans import ConanFile
+from conans.tools import mkdir
+import os
+
+class HelloConan(ConanFile):
+    name = "%s"
+    version = "1.0"
+    build_policy = "missing"
+    options = {"switch": ["1", "0"]}
+    default_options = "switch=0"
+    %s
+    
+    def build(self):
+        self.output.warn("Env var MYVAR=" + os.getenv("MYVAR", ""))
+
+    def package_info(self):
+        self.output.warn("Info from " + self.name)
+        if self.options.switch == "0": 
+            self.env_info.MYVAR = "foo"
+        else:
+            self.env_info.MYVAR = "bar"
+        
+
+'''
+        for index in range(4):
+            requires = "requires = 'Lib%s/1.0@conan/stable'" % index if index > 0 else ""
+            conanfile = conanfile_tmp % ("Lib%s" % (index + 1), requires)
+            client.save({CONANFILE: conanfile}, clean_first=True)
+            client.run("create conan/stable")
+
+        txt = "[requires]\nLib4/1.0@conan/stable"
+        client.save({CONANFILE_TXT: txt}, clean_first=True)
+        client.run("install . -o *:switch=1")
+        self.assertIn("Lib2/1.0@conan/stable: WARN: Env var MYVAR=", client.out)
```

Comment:
This assert is basically the same (or contained) in the next one

Reply:
Actually is a bug, good catch, I wanted to check Lib1, with no value. I prepended a "." to check it right.